Mathlib/Combinatorics/SimpleGraph/Basic.lean

```diff
@@ -806,4 +806,26 @@ def incidenceSetEquivNeighborSet (v : V) : G.incidenceSet v ≃ G.neighborSet v
 
 end Incidence
 
+section Subsingleton
+
+protected theorem subsingleton_iff : Subsingleton (SimpleGraph V) ↔ Subsingleton V := by
+  refine ⟨fun h ↦ ?_, fun _ ↦ Unique.instSubsingleton⟩
+  contrapose! h
+  exact instNontrivial
+
+protected theorem nontrivial_iff : Nontrivial (SimpleGraph V) ↔ Nontrivial V := by
+  refine ⟨fun h ↦ ?_, fun _ ↦ instNontrivial⟩
+  contrapose! h
+  exact Unique.instSubsingleton
+
+theorem nontrivial_of_ne_bot {G : SimpleGraph V} (h : G ≠ ⊥) : Nontrivial V := by
+  contrapose! h
+  exact SimpleGraph.subsingleton_iff.mpr h |>.allEq _ _
+
+theorem nontrivial_of_ne_top {G : SimpleGraph V} (h : G ≠ ⊤) : Nontrivial V := by
+  contrapose! h
+  exact SimpleGraph.subsingleton_iff.mpr h |>.allEq _ _
```

Comment:
I must say I don't really see the point of these two lemmas. You can obtain them both as `SimpleGraph.nontrivial_iff.2 ⟨_, _, h⟩`, which is shorter and more principled